corehq/apps/users/views/mobile/users.py

```diff
@@ -1126,6 +1128,71 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
+class DeleteCommCareUsers(BaseManageCommCareUserView):
+    urlname = 'delete_commcare_users'
+    page_title = ugettext_noop('Bulk Delete')
+    template_name = 'users/bulk_delete.html'
+
+    @property
+    def page_context(self):
+        context = self.main_context
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(),
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
+        try:
+            workbook = get_workbook(request.FILES.get('bulk_upload_file'))
+        except WorkbookJSONError as e:
+            messages.error(request, str(e))
+            return self.get(request, *args, **kwargs)
+
+        try:
+            sheet = workbook.get_worksheet()
+        except WorksheetNotFound:
+            messages.error(request, _("Workbook has no worksheets"))
+            return self.get(request, *args, **kwargs)
+
+        try:
+            usernames = [format_username(row['username'], request.domain) for row in sheet]
+        except KeyError:
+            messages.error(request, _("No users found. Please check your file contains a 'username' column."))
+            return self.get(request, *args, **kwargs)
```

Comment:
Can't the above three validations go in the form instead?

Reply:
I don't think this level of validation belongs in the form

Reply:
why so @snopoke ?
I was thinking
1. nice to have all validations at one place
2. the uploaded file is an attribute of the form so it should be validated like clean_field.

Reply:
Leaving this in the view, I think it's more common in HQ to process file contents in the view, not the form where it was uploaded (probably because we often mix validation & processing).